lib/puppet/type/cisco_service.rb

```diff
@@ -0,0 +1,117 @@
+#
+# Manages the version of Cisco Image running on a device.
+#
+# Copyright (c) 2016-2017 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_service) do
+  @doc = "Manages the version of Cisco Image running on a device.
+
+  ```
+  cisco_service {\"<version>\":
```

Comment:
While services is the larger bucket for router utilites supported by puppet, it would be better to call the type `cisco_upgrade` instead of `cisco_service`.